app/vmalert/datasource/init.go

```diff
@@ -46,7 +46,7 @@ var (
 	queryStep = flag.Duration("datasource.queryStep", 5*time.Minute, "How far a value can fallback to when evaluating queries. "+
 		"For example, if -datasource.queryStep=15s then param \"step\" with value \"15s\" will be added to every query. "+
 		"If set to 0, rule's evaluation interval will be used instead.")
-	queryTimeAlignment = flag.Bool("datasource.queryTimeAlignment", true, `Whether to align "time" parameter with evaluation interval.`+
+	queryTimeAlignment = flag.Bool("datasource.queryTimeAlignment", false, `Whether to align "time" parameter with evaluation interval.`+
```

Comment:
we should explicitly mention that this flag is deprecated and what user should use instead.

Reply:
I think we also should to remove the usage of this flag. Instead, we should add a WARN log to Init function for cases when this flag was explicitly disabled by the user. The message should repeat that this flag is deprecated, has no power and what user should do instead.

Reply:
Done! Thanks!